pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -116,6 +123,26 @@ type BGPStandardCommunity string
 // +kubebuilder:validation:Pattern=`^([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5])$`
 type BGPLargeCommunity string
 
+// BGPExtendedCommunity type represents a value of the BGP Extended Communities Attribute (RFC 4360).
+type BGPExtendedCommunity struct {
+	// SubType indicates a particular sub-type of extended community.
+	//
+	// +kubebuilder:validation:Enum=RouteTarget;RouteOrigin;LinkBandwidth
+	// +kubebuilder:validation:Required
+	SubType string `json:"subType"`
+
+	// Value holds the actual value of the extended community of the given sub-type.
+	// It can be represented in one the following formats:
+	// - Two-Octet AS Specific Extended Community: 2-byte decimal number + 4-byte decimal number separated by a colon.
+	// - Four-Octet AS-Specific Extended Community: two 2-byte decimal numbers separated by a dot + 2-byte decimal number separated by a colon.
+	// - IPv4-Address-Specific Extended Community: IPv4 address + 2-byte decimal number separated by a colon.
+	// - IPv6-Address-Specific Extended Community: IPv6 address + 2-byte decimal number separated by a colon.
+	//
+	// +kubebuilder:validation:Pattern=`^([0-9]|[1-9][0-9]{1,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5])$|^([0-9]|[1-9][0-9]{1,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])\.([0-9]|[1-9][0-9]{1,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]):([0-9]|[1-9][0-9]{1,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$|^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}:([0-9]|[1-9][0-9]{1,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$|^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)):([0-9]|[1-9][0-9]{1,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$`
```

Comment:
😅

Reply:
Yeah, there would one more alternative - to have 4 different types (e.g. `TwoOctetASSpecificEC`, `FourOctetASSpecificEC`, `IPv4AddressSpecificEC`, `IPv6AddressSpecificEC`) and each of them would have its own regex. 

That would simplify the validation pattern a bit, and make its use more explicit, but would make the `BGPExtendedCommunity` a bit more complex - probably we would need a new type `BGPExtendedCommunityValue` with 4 members of each type and `OneOf` validation.

Reply:
I am okay with leaving it as is. Is it possible to add automated tests to make sure this regex is working as intended ?

Reply:
I guess that would require either parsing it from the generated yaml manifest (`pkg/k8s/apis/cilium.io/client/crds/v2alpha1/ciliumbgppeeringpolicies.yaml`), or bringing up a test apiserver and loading the CRDs in it. Or do you have a better idea?